app/views/templates/script/package_action.erb

```diff
@@ -115,9 +115,19 @@ handle_zypp_res_codes () {
       end
     end
   -%>
+  OUTFILE=$(mktemp)
+  trap "rm -f $OUTFILE" EXIT
+  set -o pipefail
   export DEBIAN_FRONTEND=noninteractive
   apt-get -y update
-  apt-get -o APT::Get::Upgrade-Allow-New="true" -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" -y <%= input("options") %> <%= action %> <%= input("package") %>
+  apt-get -o APT::Get::Upgrade-Allow-New="true" -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" -y <%= input("options") %> <%= action %> <%= input("package") %> 2>&1 | tee $OUTFILE
+  RETVAL=$?
+  if grep -q "Unable to locate" $OUTFILE; then
+    true
+  else
+    setReturnValue() { RETVAL=$1; return $RETVAL; }
+    setReturnValue $RETVAL
```

Comment:
I'm puzzled by this, why create a function that takes an argument, sets RETVAL to the value of the argument and returns the argument, only to call it with the value that is already in RETVAL? What is the trick that I'm missing?

Reply:
The return value is grabbed here: https://github.com/theforeman/foreman_remote_execution/pull/881/files/2d60047398276ee1e23c7657aba5ec4cc5655ab8#diff-e3f43610dc58afecce93eef51717baa080bcdbda143593286b46922444cfecfaL133

This is used to prevent the original return code from the "apt-get" and set it so that the line L133 uses the return code from the apt-get. 

I tried to find a better solution like a "false" + the $RETVAL from L124 but there is nothing like this.